app-example/src/main/java/com/badoo/ribs/example/network/ApiFactory.kt

```diff
@@ -11,16 +14,22 @@ import retrofit2.converter.moshi.MoshiConverterFactory
 
 object ApiFactory {
     private const val BASE_URL = "https://api.unsplash.com/"
-    fun api(isDebug: Boolean, accessKey: String, authStateStorage: AuthStateStorage): UnsplashApi =
+    fun api(
+        isDebug: Boolean,
+        accessKey: String,
+        authStateStorage: AuthStateStorage,
+        networkErrorConsumer: Consumer<NetworkError>
+    ): UnsplashApi =
         retrofit(
-            client = okhttpClient(isDebug, accessKey, authStateStorage),
+            client = okhttpClient(isDebug, accessKey, authStateStorage, networkErrorConsumer),
             baseUrl = BASE_URL
         ).create(UnsplashApi::class.java)
 
     private fun okhttpClient(
         isDebug: Boolean,
         accessKey: String,
-        authStateStorage: AuthStateStorage
+        authStateStorage: AuthStateStorage,
+        networkErrorConsumer: Consumer<NetworkError>
```

Comment:
Is this used for anything?

Reply:
Yes `UnauthorizedErrorInterceptor` publishes events to the consumer